src/common/conditional_compilation/docs/develop_cc_for_new_component.md

```diff
@@ -0,0 +1,209 @@
+    
+- Define new macro according to the actual requirement by target component
+  The same macro should be defined in the 3 modes separately, below is an example:
+
+```ruby
+    #if defined(SELECTIVE_BUILD_ANALYZER)
+
+    #define MY_SCOPE(region) OV_SCOPE(my_op, region)
+
+    #elif defined(SELECTIVE_BUILD)
+
+    #define MY_SCOPE(region) MATCHER_SCOPE_(my_op, region)
+
+    #else
+
+    #define MY_SCOPE(region)
+
+    #endif
+```
+
+### step 3: apply CC macro
+Apply the CC macro into your code region that need conditional compilation.
+An example:
+```ruby
+    shared_ptr<ov::Node> ov::op::internal::AUGRUCell::clone_with_new_inputs(const OutputVector& new_args) const {
+        My_SCOPE(internal_AUGRUCell_clone_with_new_inputs);
+        check_new_args_count(this, new_args);
+        return make_shared<AUGRUCell>(new_args.at(0),
+                                  new_args.at(1),
+                                  new_args.at(2),
+                                  new_args.at(3),
+                                  new_args.at(4),
+                                  new_args.at(5),
+                                  get_hidden_size());
+    }
+```
+   Then code region in ov::op::internal::AUGRUCell::clone_with_new_inputs() is implement conditional compilation. If in the SELECTIVE_BUILD_ANALYZER mode it was not called, then in SELECTIVE_BUILD Mode, the code region in this function will be stripped out. 
+
+
+
+## How to verify the new conditional compilation feature
+
+#### Build OpenVINO in SELECTIVE_BUILD_ANALYZER mode
+```ruby
+    cmake \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DENABLE_PROFILING_ITT=ON \
+            -DSELECTIVE_BUILD=COLLECT \
+            ..
+    make -j12
+```
+
+#### Generate and collect analyzed data
+```ruby
+    dir=`pwd`
+    cd build-x86_64/thirdparty/itt_collector
+    make
+    cd $dir
+    mkdir -p $dir/cc_data
+    cd $dir/bin/intel64/Release
+    python3.8 ../../../thirdparty/itt_collector/runtool/sea_runtool.py --bindir $OPENVINO_HOME/bin/intel64/Release -o  <cc_data_dir>/data ! ./benchmark_app -niter 1 -nireq 1 -m <your_model.xml> -d CPU
+    cd $dir
+```
+
+#### Build OpenVINO in SELECTIVE_BUILD mode
+```ruby
+    cmake \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DENABLE_LTO=ON \
+            -DSELECTIVE_BUILD=ON \
+            -DSELECTIVE_BUILD_STAT=$OPENVINO_HOME/cc_data/*.csv \
+            -DENABLE_PROFILING_ITT=OFF 
+            ..
+    make -j8
+```
+    Note: It will benefit more if build OpenVINO static library with conditional complication, you can add option of `-DBUILD_SHARED_LIBS=OFF`.
+
+#### Run test sample 
+Run test samples built in SELECTIVE_BUILD and check whether there will be any error reported.
+```ruby
+    ./benchmark_app -m <your_model.xml> -d CPU
+``` 
+
+#### Check the binaries size benefit.
+Get the binaries size and compare with master branch result(need do as step 1 to 4), check whether the binaries is smaller than master, in general 100~500 KB.
```

Comment:
It is not correct sentence, the binary size improvement depends on set of parameters: topology, dynamic or static build, OS and etc. We cannot predict the size improvement. I propose to remove the last sentence

Reply:
ok, understood, will remove it!

Reply:
done